app/components/animated_number/index.test.tsx

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fireEvent, render, waitFor} from '@testing-library/react-native';
+import React from 'react';
+import {Animated} from 'react-native';
+
+import AnimatedNumber from '.';
+
+// jest.mock('../../../node_modules/react-native/Libraries/Text/Text', () => {
+//     const Text = jest.requireActual('../../../node_modules/react-native/Libraries/Text/Text');
+
+//     // class MockText extends React.Component {
+//     //     render() {
+//     //         return React.createElement('Text', this.props, this.props.children);
+//     //     }
+//     // }
+
+//     const MockText = jest.fn((props) => (<Text
+//         {...props}
+//         onLayout={() => ({nativeEvent: {layout: {height: 10}}})}
+//     />));
+
+//     return MockText;
+// });
+
+// no longer used for testing, but left for comparison
+describe.skip('AnimatedNumber, using toMatchSnapshot', () => {
+    it('should render correctly', () => {
+        const {toJSON} = render(<AnimatedNumber animateToNumber={123}/>);
+        expect(toJSON()).toMatchSnapshot();
+    });
+
+    it('should rerender the correct number', () => {
+        const {toJSON, getByTestId, rerender} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        expect(toJSON()).toMatchSnapshot();
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: 10}}});
+
+        rerender(<AnimatedNumber animateToNumber={124}/>);
+
+        rerender(<AnimatedNumber animateToNumber={125}/>);
+
+        expect(toJSON()).toMatchSnapshot();
+    });
+});
+
+const NUMBER_HEIGHT = 10;
+
+describe('AnimatedNumber', () => {
+    // running on jest, since Animated is a native module, Animated.timing.start needs to be mocked in order to update to the final Animated.Value.
+    // Ex: 1 => 2, the Animated.Value should be -20 (from -10) after the animation is done
+    jest.spyOn(Animated, 'timing').mockImplementation((a, b) => ({
```

Comment:
timing is taking a value from one to another at a certain duration, I don't care how long but in this example, I want animation.value to become b.toalue.

The implementation could be a bit more complicated like using easing, etc.  So it might be unique in other test cases.